libs/ledger-live-common/src/families/cosmos/crypto/crypto.ts

```diff
@@ -0,0 +1,305 @@
+import { makeLRUCache } from "../../../cache";
+import {
+  CosmosDistributionParams,
+  CosmosOperationMode,
+  CosmosPool,
+  CosmosRewardsState,
+  CosmosTotalSupply,
+} from "../types";
+import network from "../../../network";
+import { getCryptoCurrencyById } from "../../../currencies";
+import type { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { parseUatomStrAsAtomNumber } from "../logic";
+
+class Crypto {
+  currency: CryptoCurrency;
+  currencyId: string;
+  lcd: string;
+  default_gas = 100000;
+  min_gasprice = 0.0025;
+  version = "v1beta1";
+  gas: {
+    [Key in CosmosOperationMode]: number;
+  } = {
+    // refer to https://github.com/chainapsis/keplr-wallet/blob/master/packages/stores/src/account/cosmos.ts#L113 for the gas fees
+    send: 80000,
+    delegate: 250000,
+    undelegate: 250000,
+    redelegate: 250000,
+    claimReward: 140000,
+    claimRewardCompound: 400000,
+  };
+  constructor(currencyId: string) {
+    if (currencyId === "osmosis") {
+      currencyId = "osmo";
+    }
+    this.currency = getCryptoCurrencyById(currencyId);
+    this.currencyId = currencyId;
+    if (currencyId === "cosmos") {
+      this.lcd = "https://cosmoshub4.coin.ledger.com";
+      this.min_gasprice = 0.025;
+    } else if (currencyId === "osmo") {
+      this.lcd = "https://osmosis.coin.ledger.com/node";
+      this.gas = {
+        send: 100000,
+        delegate: 300000,
+        undelegate: 350000,
+        redelegate: 550000,
+        claimReward: 300000,
+        claimRewardCompound: 400000,
+      };
+    } else if (currencyId === "juno") {
+      this.lcd = "https://lcd-juno.itastakers.com";
+    } else {
+      throw new Error(`${currencyId} is not supported`);
+    }
+  }
+  validatorEstimatedRate(
+    validatorCommission: number,
+    rewardsState: CosmosRewardsState
+  ): number {
+    if (this.currencyId === "cosmos") {
+      // This correction changes how inflation is computed vs. the value the network advertises
+      const inexactBlockTimeCorrection =
+        rewardsState.assumedTimePerBlock / rewardsState.averageTimePerBlock;
+      // This correction assumes a constant bonded_ratio, this changes the yearly inflation
+      const yearlyInflation = this.computeAvgYearlyInflation(rewardsState);
+      // This correction adds the fees to the rate computation
+      const yearlyFeeRate =
+        (rewardsState.averageDailyFees * 365.24) / rewardsState.totalSupply;
+      return (
+        inexactBlockTimeCorrection *
+        (yearlyInflation + yearlyFeeRate) *
+        (1 / rewardsState.actualBondedRatio) *
+        (1 - rewardsState.communityPoolCommission) *
+        (1 - validatorCommission)
+      );
+    } else if (this.currencyId === "osmo") {
+      return 0.15; // todo fix this obviously
```

Comment:
Like we talk I'm not really compliant with having a fixed
we can take a look at what keplr do :

https://github.com/chainapsis/keplr-wallet/blob/f0c2fa6e0e04c06e163f6d3a44f1164d90f76171/packages/mobile/src/screens/stake/validator-list/index.tsx

Reply:
Fixed